Cakefile

```diff
@@ -29,6 +29,24 @@ header = """
 # Used in folder names like `docs/v1`.
 majorVersion = parseInt CoffeeScript.VERSION.split('.')[0], 10
 
+# CoffeeScript supports a range of major versions of NodeJS. Major versions of
+# NodeJS differ in the features they support. CoffeeScript features which
+# depend on features which are not available across all supported NodeJS
+# versions need only be tested against versions of NodeJS which provide the
+# required feature. The purpose of the `testFilesToSkip` array is to exempt
+# files which test features not available within the NodeJS runtime currently
+# under test.
+testFilesToSkip = []
+
+# The `skipUnless` function adds file names to the `testFilesToSkip` array
+# when the feature(s) the file(s) depend on are not available.
+skipUnless = (code, names...) ->
+  if not (try new Function code)
+    testFilesToSkip = testFilesToSkip.concat names
+
+skipUnless 'async ()   => {}',                     'async.coffee'
```

Comment:
Technically, if async functions aren’t supported, we should be skipping `async.coffee` and `async_iterators.coffee` both. It’s irrelevant in this case because you have the second check below, but if we’re thinking of this code as being self-documenting—i.e., ”the files to skip when this test fails are…”—then the second argument should be `['async.coffee', 'async_iterators.coffee']`.

Also I’m not sure how scalable this style is, when we add a test that isn’t < 60 characters or so, but we can worry about that when that day comes.

Reply:
It's easy to re-factor for widely-applicable tests

```coffee
asyncTestFiles = (files...) ->
  skipUnless 'async () = {}', files...

asyncTestFiles 'async.coffee', 'async_iterator.coffee'
asyncTestFiles 'some_other_async_test.coffee'
```

Or we could get _really_ wild...

```coffee
asyncTest = 'async () -> {}'
appendDotCoffee = (s) -> s + ".coffee"

asyncPrefixes = (s) -> skipUnless asyncTest, s.split(/\s+/).map(appendCoffee)...

asyncPrefixes """
   async
  async_iterators
  some_other_async_test
"""
```

Can you tell I <3 CoffeeScript? :)

Reply:
> Can you tell I <3 CoffeeScript? :)

Indeed. 😄 We try to make the CoffeeScript codebase a “best practices” example of how to write CoffeeScript, so code golf isn’t exactly what we’re going for. 😄 I think something as simple as making the second argument an array is probably sufficient.

Reply:
That's my Perl roots showing. Sorry about that. :) As you said, we can cross that bridge when we come to it.